### PR TITLE
Fifty move rule

### PR DESCRIPTION
**Pull request by RYU-P**

### CODE REVIEW

**Review by Dillon-Roller — APPROVED** [review]
@DanielAlbl approve this too if it's good

File: chess/src/game.rs [hunk 1]
```diff
@@ -36,6 +36,7 @@ pub enum GameStatus {
     Check,
     Checkmate,
     Replay,
+    Draw,
```
Comment (DanielAlbl) [inline chess/src/game.rs:39]: So a `Stalemate` is also a draw, so maybe we either have a single status for all draws, or name this something specific to the 50 move rule.

**Review by DanielAlbl — COMMENTED** [review]
Awesome

File: chess/src/history.rs [hunk 2]
```diff
@@ -18,7 +19,21 @@ impl History {
 
     pub fn add_info(&mut self, next_state: BoardState, mv: Move) {
         self.turns.push(Turn::new(next_state, mv));
-        self.current_turn += 1
+        self.current_turn += 1;
+        let is_pawn = next_state.get_piece(&mv.to).unwrap().is_pawn();
```
Comment (DanielAlbl) [inline chess/src/history.rs:23]: So assuming promoting a pawn counts as a pawn move, the `mv.to` square would no longer be a pawn in that case. So maybe we should check if `mv.from` in the previous state was a pawn instead.

Comment (RYU-P) [reply chess/src/history.rs:23]: big brain i oversaw that. imma test this out ! thx dan

File: chess/src/game.rs (hunk 1 above)
Comment (RYU-P) [reply chess/src/game.rs:39]: ok perfectoo

**Review by DanielAlbl — APPROVED** [review]

**Pull request merged**